src/dbcparser.cpp

```diff
@@ -97,7 +97,7 @@ bool DBCParser::parse(const std::string& data) noexcept
 
     strings phrases;
     std::deque<std::string> idents, signs, ecu_tokens;
-    std::deque<std::int64_t> numbers;
+    std::deque<std::double_t> numbers;
```

Comment:
struct CANsignal has factor and other such numbers defined as float, so you should use the same here.

Reply:
Perhaps this is something we should look and DBC specification? Grammar is defined so that a number is either int or float so perhaps we should use float